Assignment6.Rmd

```diff
@@ -52,31 +52,75 @@ post(c.tree1, file = "tree1.ps", title = "MOOC") #This creates a pdf image of th
-
 ```{r}
 M2$predict1 <- predict(c.tree1, M2, type = "class")
-
 M2$predict2 <- predict(c.tree2, M2, type = "class")
 
-table(M2$certified, M2$predict1)
+table1 <- table(M2$certified, M2$predict1)
+table2 <- table(M2$certified, M2$predict2)
 
-table(M2$certified, M2$predict2)
+#Table 1 percent correct predictions
+table1[1,1]/(table1[1,1]+table1[1,2])
+table1[2,1]/(table1[2,1]+table1[2,2])
+#Tree 1 is very good at predicting
 
+#Table 2 percent correct predictions
+table2[1,1]/(table2[1,1]+table2[1,2])
+table2[2,1]/(table2[2,1]+table2[2,2])
+
+#Tree 1 predicts much better than Tree 2. It has a lower error rate. This makes sense because it has more branches.
 ```
 
 ##Part III
 
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
+#feature select = select only some variables to be included
+#or feature extract = combine variables 
 ```{r}
+rm(list = ls())
+
+data <- read.csv("student.record.csv", header = TRUE)
+data2 <- data[,4:13]
+data2<- na.omit(data2)
+
+#this tree uses HS_GPA, all SAT components, and all ACT components to predict student's gender.
+c.tree3 <- rpart(as.factor(SEX) ~ HSGPA + LAST_ACT_ENGL_SCORE + LAST_ACT_MATH_SCORE + LAST_ACT_READ_SCORE + LAST_ACT_SCIRE_SCORE + LAST_SATI_VERB_SCORE + LAST_SATI_MATH_SCORE,  method="class", data=data2)
+
+post(c.tree3, file = "tree3.ps", title = "Tree 3") #This creates a pdf image of the tree
+
+#colnames(data2)
+#printcp(c.tree3)
+
+#this tree uses just HS_GPA, SAT overall score, and ACT overall score (combined variables) to predict student's gender.
+c.tree4 <- rpart(as.factor(SEX) ~ HSGPA + LAST_ACT_COMP_SCORE + LAST_SATI_TOTAL_SCORE,  method="class", data=data2)
+post(c.tree4, file = "tree4.ps", title = "Tree 4") #This creates a pdf image of the tree
+
+#based on the trees it looks like ACT and SAT are more useful in predicting than HS_GPA.
+
+data2$predict3 <- predict(c.tree3, data2, type = "class")
+data2$predict4 <- predict(c.tree4, data2, type = "class")
+
+table1 <- table(data2$SEX, data2$predict3)
+table2 <- table(data2$SEX, data2$predict4)
+
+#Table 1 percent correct predictions
+table1[1,1]/(table1[1,1]+table1[1,2])
+table1[2,1]/(table1[2,1]+table1[2,2])
+
+#Table 2 percent correct predictions
+table2[1,1]/(table2[1,1]+table2[1,2])
+table2[2,1]/(table2[2,1]+table2[2,2])
+
+#It looks like Tree 4 is slightly better at predicting than Tree 3. This suggests overall ACT and SAT scores are more useful at predicting a student's gender than breaking it down into the test components. 
```

Comment:
You have a good understanding of model comparison. It is interesting that you use gender as the outcome variable and standardized score as predictors. This is totally fine in this class assignment, but you may want to justify this research question (why you choose these features/predictors and outcome) if you work on projects like this.